examples/crowdfunding/test/failing-tests.js

```diff
@@ -0,0 +1,437 @@
+        amountMicroAlgos: 300000,
+        lsig: lsig,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), rejectMsg);
+  });
+
+  it('should fail if goal is not met, donor tries to reclaim funds before fund close date', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+
+    // donor should be able to reclaim if goal is NOT met
+    appArgs = [stringToBytes('reclaim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs,
+        accounts: [escrow.address] //  AppAccounts
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: donor.address,
+        amountMicroAlgos: 300000,
+        lsig: lsig,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), rejectMsg);
+  });
+
+  it('should fail if creator tries to claim funds before fund end date', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    appArgs = [stringToBytes('claim')];
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: creator.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        lsig: lsig,
+        payFlags: { totalFee: 1000, closeRemainderTo: creator.address }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), rejectMsg);
+  });
+
+  it('should fail if a transaction is missing in group transaction', () => {
+    updateAndOptIn();
+    appArgs = [stringToBytes('donate')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), 'RUNTIME_ERR1008: Index out of bound');
+  });
+
+  it('should fail if transaction is signed by wrong lsig', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    const escrowProg = getProgram('wrongEscrow.teal', { APP_ID: applicationId });
```

Comment:
What happens if i don't update the crowdfunding application with the escrow address? Does that constitute the same case as this one, or is it different?

Reply:
that will be different i think, this case checks logic signature, in this case sender is escrow but logic signature is different.
adding that case as well.

Reply:
thanks